### PR TITLE
support Blockstorage Service List

### PR DESCRIPTION
**Pull request by taemin77**
I added Block Storage Service List API.

{
    "services": [
        {
            "binary": "cinder-scheduler",
            "host": "host1",
            "state": "UP",
            "status": "ENABLED",
            "updated_at": "2012-10-29T13:42:02.000000",
            "zone": "nova"
        },
        {
            "binary": "cinder-volume",
            "host": "host1",
            "state": "UP",
            "status": "ENABLED",
            "updated_at": "2012-10-29T13:42:05.000000",
            "zone": "nova"
        }
    ]
}

### CODE REVIEW

**Comment by auhlig** [comment]
There are some conflicts, due to your previous PR. Could you resolve them @taemin77?

**Comment by taemin77** [comment]
I resolved it @auhlig.

File: core-test/src/main/java/org/openstack4j/api/storage/ServiceTests.java [hunk 1]
```diff
@@ -14,26 +14,28 @@
  * 
  * @author Taemin
  */
+
 @Test(suiteName = "BlockStroageServices")
```
Comment (auhlig) [inline core-test/src/main/java/org/openstack4j/api/storage/ServiceTests.java]: Typo: BlockStroageServices -> BlockStorageService

File: core/src/main/java/org/openstack4j/api/storage/BlockStorageService.java [hunk 2]
```diff
@@ -50,9 +50,9 @@
...
-     *  The block storage services service
-     *  
-     * @return ServicesService
-     */
-	BlockStroageServicesService services();
+	 * The block storage services service
+	 * 
+	 * @return ServicesService
+	 */
+	 BlockStroageServicesService services();
```
Comment (auhlig) [inline core/src/main/java/org/openstack4j/api/storage/BlockStorageService.java]: Typo: BlockStroageServices -> BlockStorageService

File: core/src/main/java/org/openstack4j/api/storage/ext/BlockStroageServicesService.java [hunk 3]
```diff
@@ -4,22 +4,19 @@
...
 import org.openstack4j.model.storage.block.ext.Service;
 
-
-
 /**
  * API which supports the "os-services" extension.
  *
  * @author Taemin
  */
 public interface BlockStroageServicesService {
```
Comment (auhlig) [inline core/src/main/java/org/openstack4j/api/storage/ext/BlockStroageServicesService.java]: Typo: BlockStroageServices -> BlockStorageService

File: core/src/main/java/org/openstack4j/openstack/provider/DefaultAPIProvider.java [hunk 4]
```diff
@@ -479,7 +479,6 @@ public void initialize() {
         bind(AlarmAodhService.class, AlarmAodhServiceImpl.class);
         bind(ServicesService.class, ServicesServiceImpl.class);
         bind(BlockStroageServicesService.class, BlockStorageServicesServiceImpl.class);
```
Comment (auhlig) [inline core/src/main/java/org/openstack4j/openstack/provider/DefaultAPIProvider.java]: Typo: BlockStroageServices -> BlockStorageService

File: core/src/main/java/org/openstack4j/openstack/storage/block/internal/BlockStorageServiceImpl.java [hunk 5]
```diff
@@ -77,7 +77,6 @@ public BlockVolumeBackupService backups() {
 	@Override
 	public BlockStroageServicesService services() {
 		return Apis.get(BlockStroageServicesService.class);
```
Comment (auhlig) [inline core/src/main/java/org/openstack4j/openstack/storage/block/internal/BlockStorageServiceImpl.java]: Typo: BlockStroageServices -> BlockStorageService

File: core/src/main/java/org/openstack4j/openstack/storage/block/internal/BlockStorageServicesServiceImpl.java [hunk 6]
```diff
@@ -6,20 +6,19 @@
 import org.openstack4j.model.storage.block.ext.Service;
 import org.openstack4j.openstack.storage.block.domain.ext.ExtService.Services;
 
-
 /**
  * Block Storage Services service provides CRUD capabilities for Cinder service(s).
  *
  * @author Taemin
  */
 public class BlockStorageServicesServiceImpl extends BaseBlockStorageServices implements BlockStroageServicesService {
```
Comment (auhlig) [inline core/src/main/java/org/openstack4j/openstack/storage/block/internal/BlockStorageServicesServiceImpl.java]: Typo: BlockStroageService -> BlockStorageServic

**Comment by taemin77** [comment]
Thank you for checking code.

**Comment by auhlig** [comment]
Could you also address the other typos @taemin77? Could be done quickly by a search&replace.

**Comment by taemin77** [comment]
I did work.
Could you check codes again?

File: core/src/main/java/org/openstack4j/api/storage/BlockStorageService.java [hunk 7]
```diff
@@ -1,6 +1,6 @@
 package org.openstack4j.api.storage;
 
-import org.openstack4j.api.storage.ext.BlockStroageServicesService;
+import org.openstack4j.api.storage.ext.BlockStroageServiceService;
```
Comment (auhlig) [inline core/src/main/java/org/openstack4j/api/storage/BlockStorageService.java]: I can still see multiple `BlockStroageServiceService` in this commit.

File: core/src/main/java/org/openstack4j/openstack/provider/DefaultAPIProvider.java [hunk 8]
```diff
@@ -290,7 +298,15 @@
 import org.openstack4j.openstack.tacker.internal.VimServiceImpl;
 import org.openstack4j.openstack.tacker.internal.VnfServiceImpl;
 import org.openstack4j.openstack.tacker.internal.VnfdServiceImpl;
-import org.openstack4j.openstack.telemetry.internal.*;
+import org.openstack4j.openstack.telemetry.internal.AlarmAodhServiceImpl;
```
Comment (auhlig) [inline core/src/main/java/org/openstack4j/openstack/provider/DefaultAPIProvider.java]: revert imports

File: core/src/main/java/org/openstack4j/api/storage/BlockStorageService.java (hunk 7 above)
Comment (taemin77) [inline core/src/main/java/org/openstack4j/api/storage/BlockStorageService.java]: ok

File: core/src/main/java/org/openstack4j/openstack/provider/DefaultAPIProvider.java (hunk 8 above)
Comment (taemin77) [inline core/src/main/java/org/openstack4j/openstack/provider/DefaultAPIProvider.java]: ok

**Comment by taemin77** [comment]
openstack4j is the first time I cooperate on github.
so there is some difficulty to check git status.
what could I do to finish PR now?

File: core/src/main/java/org/openstack4j/openstack/storage/block/internal/BlockStorageServiceImpl.java [hunk 9]
```diff
@@ -75,9 +75,8 @@ public BlockVolumeBackupService backups() {
 	 * {@inheritDoc}
 	 */
 	@Override
-	public BlockStroageServicesService services() {
-		return Apis.get(BlockStroageServicesService.class);
-		
+	public BlockStroageServiceService services() {
```
Comment (auhlig) [inline core/src/main/java/org/openstack4j/openstack/storage/block/internal/BlockStorageServiceImpl.java]: BlockStroageServiceService -> BlockStorageServiceService (`roa`-> `ora`)

File: core/src/main/java/org/openstack4j/openstack/provider/DefaultAPIProvider.java [hunk 10]
```diff
@@ -480,8 +496,8 @@ public void initialize() {
         bind(TelemetryAodhService.class,TelemetryAodhServiceImpl.class);
         bind(AlarmAodhService.class, AlarmAodhServiceImpl.class);
         bind(ServicesService.class, ServicesServiceImpl.class);
+        bind(BlockStroageServiceService.class, BlockStorageServiceServiceImpl.class);
```
Comment (auhlig) [inline core/src/main/java/org/openstack4j/openstack/provider/DefaultAPIProvider.java]: BlockStroageServiceService.class -> BlockStorageServiceService.class (`roa`-> `ora`)

File: core/src/main/java/org/openstack4j/openstack/provider/DefaultAPIProvider.java [hunk 11]
```diff
@@ -140,13 +140,21 @@
 import org.openstack4j.api.storage.ObjectStorageObjectService;
 import org.openstack4j.api.storage.ObjectStorageService;
 import org.openstack4j.api.storage.SchedulerStatsGetPoolService;
-import org.openstack4j.api.storage.ext.BlockStroageServicesService;
+import org.openstack4j.api.storage.ext.BlockStroageServiceService;
```
Comment (auhlig) [inline core/src/main/java/org/openstack4j/openstack/provider/DefaultAPIProvider.java]: BlockStroageServiceService -> BlockStorageServiceService (`roa`-> `ora`)

File: core/src/main/java/org/openstack4j/api/storage/ext/BlockStroageServiceService.java [hunk 12]
```diff
@@ -0,0 +1,22 @@
...
+import java.util.List;
+
+import org.openstack4j.model.storage.block.ext.Service;
+
+/**
+ * API which supports the "os-services" extension.
+ *
+ * @author Taemin
+ */
+public interface BlockStroageServiceService {
```
Comment (auhlig) [inline core/src/main/java/org/openstack4j/api/storage/ext/BlockStroageServiceService.java]: BlockStroageServiceService -> BlockStorageServiceService (`roa`->`ora`)

File: core/src/main/java/org/openstack4j/api/storage/BlockStorageService.java [hunk 13]
```diff
@@ -50,9 +50,9 @@
...
-     *  The block storage services service
-     *  
-     * @return ServicesService
-     */
-	BlockStroageServicesService services();
+	 * The block storage service service
+	 * 
+	 * @return ServiceService
+	 */
+	 BlockStroageServiceService services();
```
Comment (auhlig) [inline core/src/main/java/org/openstack4j/api/storage/BlockStorageService.java]: BlockStroageServiceService -> BlockStorageServiceService (`roa`->`ora`)

File: core-test/src/main/java/org/openstack4j/api/storage/ServiceTests.java [hunk 14]
```diff
@@ -14,26 +14,28 @@
  * 
  * @author Taemin
  */
-@Test(suiteName = "BlockStroageServices")
+
+@Test(suiteName = "BlockStroageService")
```
Comment (auhlig) [inline core-test/src/main/java/org/openstack4j/api/storage/ServiceTests.java]: BlockStroageService -> BlockStorageService (`roa`->`ora`)

File: core/src/main/java/org/openstack4j/openstack/provider/DefaultAPIProvider.java [hunk 15]
```diff
@@ -140,13 +140,21 @@
...
 import org.openstack4j.api.tacker.TackerService;
 import org.openstack4j.api.tacker.TackerServiceImpl;
 import org.openstack4j.api.tacker.VimService;
 import org.openstack4j.api.tacker.VnfService;
 import org.openstack4j.api.tacker.VnfdService;
-import org.openstack4j.api.telemetry.*;
+import org.openstack4j.api.telemetry.AlarmAodhService;
```
Comment (auhlig) [inline core/src/main/java/org/openstack4j/openstack/provider/DefaultAPIProvider.java]: please revert imports

File: core/src/main/java/org/openstack4j/openstack/storage/block/internal/BlockStorageServiceImpl.java [hunk 16]
```diff
@@ -7,10 +7,10 @@
 import org.openstack4j.api.storage.BlockVolumeService;
 import org.openstack4j.api.storage.BlockVolumeSnapshotService;
 import org.openstack4j.api.storage.CinderZoneService;
+import org.openstack4j.api.storage.SchedulerStatsGetPoolService;
+import org.openstack4j.api.storage.ext.BlockStroageServiceService;
```
Comment (auhlig) [inline core/src/main/java/org/openstack4j/openstack/storage/block/internal/BlockStorageServiceImpl.java]: BlockStroageServiceService -> BlockStorageServiceService (`roa`->`ora`)

File: core/src/main/java/org/openstack4j/openstack/storage/block/internal/BlockStorageServiceServiceImpl.java [hunk 17]
```diff
@@ -0,0 +1,24 @@
...
+import org.openstack4j.api.storage.ext.BlockStroageServiceService;
+import org.openstack4j.model.storage.block.ext.Service;
+import org.openstack4j.openstack.storage.block.domain.ext.ExtService.Services;
+
+/**
+ * Block Storage Services service provides CRUD capabilities for Cinder service(s).
+ *
+ * @author Taemin
+ */
+public class BlockStorageServiceServiceImpl extends BaseBlockStorageServices implements BlockStroageServiceService {
```
Comment (auhlig) [inline core/src/main/java/org/openstack4j/openstack/storage/block/internal/BlockStorageServiceServiceImpl.java]: BlockStroageServiceService -> BlockStorageServiceService (`roa`->`ora`)

File: core/src/main/java/org/openstack4j/openstack/provider/DefaultAPIProvider.java (hunk 15 above)
Comment (taemin77) [inline core/src/main/java/org/openstack4j/openstack/provider/DefaultAPIProvider.java]: this code is not same with my code.

**Comment by auhlig** [comment]
Well done @taemin77! Thanks for all the work.

**Pull request merged**